sklearn/utils/tests/test_validation.py

```diff
@@ -417,6 +420,15 @@ def test_check_array_dtype_warning():
     assert_equal(X_checked.format, 'csr')
 
 
+def test_check_array_warn_on_dtype_deprecation():
+    X = np.asarray([[0.0], [1.0]])
+    Y = np.asarray([[2.0], [3.0]])
+    with pytest.warns(DeprecationWarning,
+                      match="'warn_on_dtype' is deprecated"):
+        check_array(X, warn_on_dtype=True)
+        check_X_y(X, Y, warn_on_dtype=True)
```

Comment:
Please separate these 2 calls:

```py
    with pytest.warns(DeprecationWarning,
                      match="'warn_on_dtype' is deprecated"):
        check_array(X, warn_on_dtype=True)
    with pytest.warns(DeprecationWarning,
                      match="'warn_on_dtype' is deprecated"):
        check_X_y(X, Y, warn_on_dtype=True)
```

else there is no way to know which one raises a warning

Reply:
done